lib/ansible/modules/cloud/pubnub/pubnub_blocks.py

```diff
@@ -0,0 +1,2090 @@
+        :type block:        dict
+        :param block:       PubNub service response with information about particular block. 'None' possible in case if
+                            there is request to create new block.
+        :type name:         str
+        :param name:        Reference on name of block which should be created.
+        :type description:  str
+        :param description: Reference on bew block description.
+        :rtype:  PubNubBlock
+        :return: Initialized block model.
+        """
+        super(PubNubBlock, self).__init__()
+        self._module = module
+        self._changed = False
+        self._api = api_client
+        self._keyset = keyset
+        self._orig_service_data = None
+        self._block_data = None
+        self._event_handlers = dict()
+        self._should_delete = False
+        if not _is_empty(block):
+            self._process_block_data(block)
+        else:
+            self.name = name
+            self.description = description
+
+    @property
+    def keyset(self):
+        """Stores reference on keyset for which block has been created.
+
+        :rtype:  PubNubKeyset
+        :return: Reference on keyset model.
+        """
+        return self._keyset
+
+    @property
+    def payload(self):
+        """Stores reference on block data structured as it required by PubNub service.
+
+        :rtype:  dict
+        :return: Block information in format which is known to PubNub service.
+        """
+        payload = dict()
+        if not _is_empty(self._orig_service_data):
+            payload.update(key_id=self.keyset.uid, block_id=self.uid, name=self.name, description=self.description)
+        if not _is_empty(self._block_data):
+            if not _is_empty(self._block_data.get('name')):
+                payload['name'] = self._block_data['name']
+            if not _is_empty(self._block_data.get('description')):
+                payload['description'] = self._block_data['description']
+        if _is_empty(payload.get('description')):
+            payload['description'] = 'New block'
+
+        return payload
+
+    @property
+    def uid(self):
+        """Stores reference on unique block's identifier.
+
+        :rtype:  int
+        :return: Reference on unique block's identifier or 'None' in case if block model configuration not completed.
+        """
+        uid = None
+        if not _is_empty(self._orig_service_data) and not _is_empty(self._orig_service_data.get('id')):
+            uid = self._orig_service_data['id']
+
+        return uid
+
+    @property
+    def name(self):
+        """Stores reference on block's name.
+
+        :rtype:  str
+        :return: Reference on block's name or 'None' in case if block model configuration not completed.
+        """
+        data = self._orig_service_data if not _is_empty(self._orig_service_data) else self._block_data
+
+        return data['name'] if not _is_empty(data) and not _is_empty(data.get('name')) else None
```

Comment:
Instead, you can use `return data.get('name')` and it will return None if name is undefined.